Parse: updated `TriggerRequest` interface, added missing `afterFind` trigger, fixed `beforeFind` signature

virtualtoy opened the pull request:
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:  test for this property exists here ('test afterSave get original object on update'): <<https://github.com/parse-community/parse-server/blob/master/spec/ParseAPI.spec.js>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

jaeggerr commented:
I think that "original" should be moved to the TriggerRequest interface as stated in the official Parse API documentation.
http://parseplatform.org/Parse-SDK-JS/api/v1.11.1/Parse.Cloud.html (end of page).

virtualtoy commented:
@jaeggerr updated `TriggerRequest` to match official Parse docs. Also added missing `afterFind` trigger and related interfaces. Fixed `beforeFind` signature.

jaeggerr commented:
Looks good to me, thank you.
